pkg/queue/delay_test.go

```diff
@@ -112,6 +129,37 @@ func TestDQ_sanity_check(t *testing.T) {
 	}
 }
 
+func TestDQ_Offer(t *testing.T) {
+	t.Parallel()
+	assert := assert.New(t)
+
+	dq := NewDelayQueue()
+	delay := time.Second
+
+	added := dq.Offer(&testjob{})
+	if added {
+		t.Fatalf("offered job without deadline added")
```

Comment:
Tiny nit: Could we change the log message to something like "DelayQueue should not accept job without deadline"